### PR TITLE
DROOLS-2554 allow variance accumulate to survive NaN on reverse

### PR DESCRIPTION
**Pull request by jgoldsmith613**
allow variance accumulate to survive a reverse when only one fact has been added
@ge0ffrey

### CODE REVIEW

File: drools-core/src/test/java/org/drools/core/base/accumulators/VarianceNaNTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package org.drools.core.base.accumulators;
+
+import org.drools.core.base.accumulators.VarianceAccumulateFunction.VarianceData;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+
+public class VarianceNaNTest {
```
Comment (baldimir) [inline drools-core/src/test/java/org/drools/core/base/accumulators/VarianceNaNTest.java]: Please format the code with our code style conventions [1]. 

[1] https://github.com/baldimir/droolsjbpm-build-bootstrap/tree/master/ide-configuration

File: drools-core/src/test/java/org/drools/core/base/accumulators/VarianceNaNTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	
+	@Test
+	public void shouldNotProduceNaNAfterBackout(){
+		VarianceAccumulateFunction varianceAccumulateFunction = new VarianceAccumulateFunction();
+		VarianceData data = varianceAccumulateFunction.createContext();
+		varianceAccumulateFunction.init(data);
+		
+		//Before being initialized result is NaN.
+		assertEquals(Double.NaN, varianceAccumulateFunction.getResult(data), 0);
+
```
Comment (baldimir) [inline drools-core/src/test/java/org/drools/core/base/accumulators/VarianceNaNTest.java]: Nitpicking: Unnecessary empty lines.

File: drools-core/src/test/java/org/drools/core/base/accumulators/VarianceNaNTest.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+		assertEquals(0.0d, varianceAccumulateFunction.getResult(data), .001d);
+		
+		//should be back to NaN after backout
+		varianceAccumulateFunction.reverse(data, value);
+		assertEquals(Double.NaN, varianceAccumulateFunction.getResult(data), 0);
+		
+		//should be zero after adding number back
+		varianceAccumulateFunction.accumulate(data, value);		
+		assertEquals(0.0d, varianceAccumulateFunction.getResult(data), .001d);
+
```
Comment (baldimir) [inline drools-core/src/test/java/org/drools/core/base/accumulators/VarianceNaNTest.java]: Nitpicking: Unnecessary empty lines.

**Review by baldimir — CHANGES_REQUESTED** [review]
Hi, thanks for contribution, I added few minor comments. The final word on this should be done probably by @mariofusco.

**Comment by ge0ffrey** [comment]
ok to test

File: drools-core/src/test/java/org/drools/core/base/accumulators/VarianceNaNTest.java (hunk 1 above)
Comment (ge0ffrey) [reply drools-core/src/test/java/org/drools/core/base/accumulators/VarianceNaNTest.java]: Contributors don't need to follow our code style to the letter. Only paid core employees need to. Whoever merges this, is responsible for reformatting to our code style. So, it's good to point this out, but it's no reason to not approve the PR.

Comment (baldimir) [reply drools-core/src/test/java/org/drools/core/base/accumulators/VarianceNaNTest.java]: Ok, thanks for clarification.

**Pull request merged**